content/en/docs/Reference/dependency-resolution.md

```diff
@@ -0,0 +1,351 @@
+    
+    classDef foo fill:#8addf2,stroke:#333,stroke-width:4px;
+    classDef bar fill:#ffcc26,stroke:#333,stroke-width:4px;
+    classDef baz fill:#ff7452,stroke:#333,stroke-width:4px;
+
+    class e,ec foo;
+    class v,vs bar;
+    
+    linkStyle default fill:none,stroke-width:2px;
+{{</mermaid>}}
+
+Installing `BarOperator` with OLM will automatically install `FooOperator` as well - assuming there are no other conflicts. 
+
+This is made possible with two types of data:
+
+ - **Properties** - typed metadata about the operator that constutes the public interface for it in the dependency resolver. Examples include the GVKs of the APIs provided by the operator and the SemVer version of the operator
+ - **Constraints** or **Dependencies** - an operator's requirements that should be satisfied by other operators (that may or may not have already been installed) or the target cluster. These act as queries / filters over all available operators and constrain the selection when resolving or installing. Examples include requiring a specific API to be available on the cluster, or expecting a particular operator with a particular version to be installed.
+
+
+Under the hood, these properties and constraints are converted into a system of boolean formulas and handed to a [SAT-solver](https://en.wikipedia.org/wiki/Boolean_satisfiability_problem), which does the heavy lifting of determining what operators should be installed.
+
+## Declaring Properties
+
+Properties cannot be directly declared by an operator author. Instead, they are derived from the existing metadata provided in the CSV.
+
+All operators in a catalog will have these properties:
+
+ - `olm.package` - includes the name of the package and the version of the operator
+ - `olm.gvk` - one property is present for each "provided" API from the ClusterServiceVersion
+
+A future release of OLM will support explicit property creation.
+
+## Declaring Dependencies
+
+Dependencies are declared by including a `dependencies.yaml` file in the `metadata` directory of the operator bundle. For more information on bundles and the their format, see the [bundle docs](https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md).
+
+Currently, only two types of constraints are supported:
+
+ - `olm.gvk` - declare a requirement on an API
+ - `olm.package` - declare a requirement on a specific package / version range
+
+
+Here's an example of a `dependencies.yaml`:
+
+```yaml
+dependencies:
+- type: olm.package
+  value:
+    packageName: prometheus
+    version: ">0.27.0"
+- type: olm.gvk
+  value:
+    group: etcd.database.coreos.com
+    kind: EtcdCluster
+    version: v1beta2
+```
+
+***Note***: The `version` field above follows the [SemVer 2.0 Spec](https://semver.org/) for version ranges, and specifically uses [blang/semver](https://github.com/blang/semver) to parse.
+
+The resolver sees these constraints as (assuming this is for `FooOperator v1.0.0`):
```

Comment:
Is it better to talk about installing the `Bar` operator here instead of the foo operator since that's what we've been discussing so far in the doc?